timeseries/src/autogluon/timeseries/splitter.py

```diff
@@ -181,8 +181,10 @@ def _split(
             validation_dataframes.append(next_val_dataframe)
 
         train_data = ts_dataframe.slice_by_timestep(slice(None, -prediction_length))
+        val_data = pd.concat(validation_dataframes)
```

Comment:
cool. am I understanding correctly that `slice` preserves `_cached_freq` but `concat` doesn't? And how did this problem surface? didn't freq inference work after concatenation?

Reply:
Yes, `_cached_freq` isn't preserved by concatenation. 

The source of the problem are datasets where some time series are really short. After we "slice off" the last `prediction_length` time steps, freq inference stops working (some series only have 2-3 time steps), so `freq` returns `None` and this led to problems downstream (`get_seasonality` didn't handle the case where `freq is None`).